### PR TITLE
DryRun for migrator

### PR DESCRIPTION
**Pull request by defool**
### What did this pull request do?
Print SQL in AutoMigrate with DryRun option.

### User Case Description
As mentioned in #3545.

### CODE REVIEW

**Review by daheige — APPROVED** [review]

File: migrator/migrator.go [hunk 1]
```diff
@@ -92,14 +105,19 @@ func (m Migrator) FullDataTypeOf(field *schema.Field) (expr clause.Expr) {
...
 func (m Migrator) AutoMigrate(values ...interface{}) error {
 	for _, value := range m.ReorderModels(values, true) {
-		tx := m.DB.Session(&gorm.Session{})
-		if !tx.Migrator().HasTable(value) {
-			if err := tx.Migrator().CreateTable(value); err != nil {
+		queryTx := m.DB.Session(&gorm.Session{})
+		execTx := queryTx
+		if m.DB.DryRun {
+			queryTx.DryRun = false
+			execTx = m.DB.Session(&gorm.Session{Logger: &printSQLLogger{}})
```
Comment (jinzhu) [inline migrator/migrator.go:112]: the `printSQLLogger` should reuse the current logger to print the running SQL?

Comment (defool) [reply migrator/migrator.go:112]: The existing logger print some tracing text, so we can not copy the output to execute it.
I think a new logger `printSQLLogger` is necessary.

Comment (defool) [reply migrator/migrator.go:112]: Any other suggestion on this PR? @jinzhu

**Comment by zchenyu** [comment]
Any progress on this PR? This would be a great feature!

**Review by jinzhu — APPROVED** [review]

**Pull request merged**